src/host/tcp.h

```diff
@@ -608,6 +608,12 @@ namespace asynchost
       {
         static_cast<TCPImpl*>(req->data)->on_resolved(req, rc);
       }
+      else
+      {
+        // The TCPImpl that submitted the request has been destroyed, but we
+        // need to clean up the request object.
+        delete req;
```

Comment:
@wintersteiger is removing this fixing a double free rather a leak? Am I missing something?

Reply:
No, it fixes a leak. `req->data` usually holds a pointer to the owning `TCPImpl` and when that gets deleted, the pointers pointing to it a reset to null. The request objects are deleted in the callbacks, but when `req->data` is null we don't call the callbacks anymore, so we need to delete the object here.

Reply:
Yep, you're right of course. I'm so tired that I'd interpreted this as code being removed rather than added. Sorry!

Reply:
No worries, thank for confirming!